geowebcache/core/src/main/java/org/geowebcache/config/DefaultingConfiguration.java

```diff
@@ -2,16 +2,91 @@
 
 import org.geowebcache.layer.TileLayer;
 
+import java.util.Collection;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
 /**
  * Applies default values to a layer that has not been fully initialized.
  *
  */
 public interface DefaultingConfiguration extends BaseConfiguration {
 
     /**
-     * TileLayerConfiguration objects lacking their own defaults can delegate to this
+     * TileLayerConfiguration objects lacking their own defaults can delegate to this.
+     *
      * @param layer
      */
     void setDefaultValues(TileLayer layer);
```

Comment:
This should be the only method needed in the interface. All the other methods only belong to TileLayerConfiguration.
Also, the javadoc doesn't do a particularly good job of explaining what the method does right now.
(TileLayerConfigurations will call this method to set up the default values for newly created/added tile layers - it may be useful to examine somewhere this actually gets called to get a better picture).